gateway/ln-gateway/src/actor_fork.rs

```diff
@@ -42,7 +53,110 @@ impl GatewayActor {
             Err(e) => warn!("Failed to connect with federation: {}", e),
         }
 
-        Ok(Self { client })
+        let actor = Self {
+            client,
+            lnrpc,
+            task_group,
+        };
+
+        actor.subscribe_htlcs().await?;
+
+        Ok(actor)
+    }
+
+    async fn subscribe_htlcs(&self) -> Result<()> {
+        let actor = self.to_owned();
+        let lnrpc = self.lnrpc.to_owned();
+        let short_channel_id = self.client.config().mint_channel_id;
+        let mut tg = self.task_group.clone();
```

Comment:
IMHO, this seems kind of weird to always (saw this now multiple times in the `GatewayActor`) clone required variables, especially cloning `self`. 
I'd love to avoid this if possible. 
For instance, we should be able to take a `&mut self` in the method itself, so we can just use `self.task_group.spawn()` and don't need to clone it. 
On the other hand, for the `lnrpc` we can create a clone (that is also just a copy of the `Arc`) which can be passed into the `task_group.spawn` closure. 
 
something like 
```rust
{ // open scope for variables moved into the task group closure
     let lnrpc_copy = self.lnrpc.clone();
     self.task_group.spawn(....) {
        ....
        lnrpc_copy.complete_htlc(...)
    }

} // end of scope

Reply:
ack. using `to_owned` as a safer copy method. Recommendation from @dpc :)